drivers/azure/azure.go

```diff
@@ -28,7 +28,7 @@ const (
 	defaultAzureLocation        = "westus"
 	defaultSSHUser              = "docker-user" // 'root' not allowed on Azure
 	defaultDockerPort           = 2376
-	defaultAzureImage           = "canonical:UbuntuServer:18.04-LTS:latest"
+	defaultAzureImage           = "canonical:ubuntu-24_04-lts:server-gen1:latest"
```

Comment:
What is the specific reason we're going with `canonical:ubuntu-24_04-lts:server-gen1:latest` and not `Canonical:ubuntu-24_04-lts:server:latest`? Also, should it be `canonical` or `Canonical` ?

Reply:
The reason for choosing `canonical:ubuntu-24_04-lts:server-gen1:latest` over `Canonical:ubuntu-24_04-lts:server:latest` is to avoid potential issues during VM creation. Specifically, when using the latter image, we may encounter the following challenges:
1. image `Canonical:ubuntu-24_04-lts:server:latest`  is not supported for creating a Virtual Machine with unmanaged disks, Please refer to the disallowed VM disk types at https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages/get#vmdisktypes.
2. The selected VM size `Standard_D2_v2` cannot boot Hypervisor Generation '2'.

The gen-1 image (canonical:ubuntu-24_04-lts:server-gen1:latest) was chosen because it aligns with our current configuration using unmanaged disks and the `Standard_D2_v2` VM size, ensuring compatibility. Additionally, the previous image we used, `canonical:UbuntuServer:18.04-LTS:latest`, was also a Gen-1 image, further justifying the decision to stick with Gen-1 for consistency...

Besides, Azure is deprecating unmanaged disks by September 30, 2025, in favor of managed disks for better scalability and reliability. https://learn.microsoft.com/en-us/azure/virtual-machines/unmanaged-disks-deprecation.

Lastly, regarding the naming convention, both `canonical` and `Canonical` refer to the same publisher, though lowercase (canonical) is typically used in this context.